modules/partition-replicator/src/test/java/org/apache/ignite/internal/partition/replicator/PartitionReplicaLifecycleManagerTest.java

```diff
@@ -260,4 +266,53 @@ void testStopOrder() throws NodeStoppingException {
         inOrder.verify(zoneResourcesManager, timeout(1_000)).destroyZonePartitionResources(zonePartitionId);
         inOrder.verify(replicaManager, timeout(1_000)).destroyReplicationProtocolStorages(zonePartitionId, false);
     }
+
+    @Test
+    void producesEventsOnPartitionRestart() {
+        var beforeReplicaStoppedFuture = new CompletableFuture<LocalPartitionReplicaEventParameters>();
+        var afterReplicaStoppedFuture = new CompletableFuture<LocalPartitionReplicaEventParameters>();
+
+        partitionReplicaLifecycleManager.listen(BEFORE_REPLICA_STOPPED, EventListener.fromConsumer(beforeReplicaStoppedFuture::complete));
+        partitionReplicaLifecycleManager.listen(AFTER_REPLICA_STOPPED, EventListener.fromConsumer(params -> {
+            beforeReplicaStoppedFuture.thenRun(() -> afterReplicaStoppedFuture.complete(params));
```

Comment:
Why is 'after' future chained to 'before' future?

Reply:
The original idea was to guarantee that `before` was fired before `after`, but now I understood that it makes no sense)